Achievements: Add customizable sound effects

kamfretoz opened the pull request:
### Description of Changes
<!-- Brief description or overview on what was changed in the PR -->
This PR adds customization options for RCheevos notification/unlock sound effects.

At the moment i've made the file picker accept only ~~MP3 and~~ WAV files.

Preview (Sound On):

https://github.com/user-attachments/assets/fe352407-3d22-42e5-b704-00e1e7b1f149

### Rationale behind Changes
<!-- Why were these changes made?  What problem does it solve / area does it improve? -->
An often requested feature and a nice to have customization.

### Suggested Testing Steps
<!-- If applicable, including examples you've already tested with / recommendations for how to test further is very helpful! -->
Test adding and resetting sounds, as well as test fresh install and portable mode.

kamfretoz commented:
I'll have to look into adding scroll area to the achievement widget, i hate how clunky Qt Designer is.

JordanTheToaster reviewed (approved):
Working on Windows with an existing install and a clean potable install.

kamfretoz commented:
I've moved it to the achievement section.

![image](https://github.com/user-attachments/assets/fdb21b1d-2642-499f-a5d8-e88c26823bd7)

spaceage64 commented:
I like this feature as I was a fan of the old sound that was used (the Xbox 360 one) and would be cool to restore it int his way. But why restrict this to wav only?

JordanTheToaster commented:
> I like this feature as I was a fan of the old sound that was used (the Xbox 360 one) and would be cool to restore it int his way. But why restrict this to wav only?

Changing it to anything else is out of scope for this PR.

spaceage64 commented:
That was a quick reply :) Anyway, I think at the very least mp3 would be a convenient addition. One more thing, I see there are buttons for "Browse..." and "Open...". I think as a user it is unclear what the distinction here is exactly. I guess I would just click browse to pick a file and it'd work, but then I'd still be unsure what the "Open..." button would be there for.

kamfretoz commented:
> I'd still be unsure what the "Open..." button would be there for.

It opens the audio file you choose.

kamfretoz commented:
Added the ability to disable each sound individually and changed the open button to preview so its more clear.

![image](https://github.com/user-attachments/assets/fd388724-f318-48ca-b1f5-8ad179b40ea1)
